ipykernel/iostream.py

```diff
@@ -199,6 +206,8 @@ def _setup_pipe_in(self):
 
     async def _handle_pipe_msgs(self):
         """handle pipe messages from a subprocess"""
+        # create async wrapper within coroutine
+        self._async_pipe_in1 = zmq.asyncio.Socket(self._pipe_in1)
```

Comment:
Just wondering why it is different than line 170 when there is `Socket.shadow()`?